sdk/search/azure-search-documents/README.md

```diff
@@ -113,61 +168,258 @@ SearchAsyncClient searchAsyncClient = new SearchClientBuilder()
+An Azure Cognitive Search service contains one or more indexes that provide
+persistent storage of searchable data in the form of JSON documents.  _(If
+you're brand new to search, you can make a very rough analogy between
+indexes and database tables.)_  The Azure.Search.Documents client library
+exposes operations on these resources through two main client types.
+
+* `SearchClient` helps with:
+  * [Searching](https://docs.microsoft.com/azure/search/search-lucene-query-architecture)
+    your indexed documents using
+    [rich queries](https://docs.microsoft.com/azure/search/search-query-overview)
+    and [powerful data shaping](https://docs.microsoft.com/azure/search/search-filters)
+  * [Autocompleting](https://docs.microsoft.com/rest/api/searchservice/autocomplete)
+    partially typed search terms based on documents in the index
+  * [Suggesting](https://docs.microsoft.com/rest/api/searchservice/suggestions)
+    the most likely matching text in documents as a user types
+  * [Adding, Updating or Deleting Documents](https://docs.microsoft.com/rest/api/searchservice/addupdate-or-delete-documents)
+    documents from an index
+
+* `SearchIndexClient` allows you to:
+  * [Create, delete, update, or configure a search index](https://docs.microsoft.com/rest/api/searchservice/index-operations)
+  * [Declare custom synonym maps to expand or rewrite queries](https://docs.microsoft.com/rest/api/searchservice/synonym-map-operations)
+  * Most of the `SearchServiceClient` functionality is not yet available in our current preview
+
+* `SearchIndexerClient` allows you to:
+  * [Start indexers to automatically crawl data sources](https://docs.microsoft.com/rest/api/searchservice/indexer-operations)
+  * [Define AI powered Skillsets to transform and enrich your data](https://docs.microsoft.com/rest/api/searchservice/skillset-operations)
 
 ## Examples
 
-### Create an index
+#### Use `SearchDocument` like a dictionary
 
-Create Index using `searchIndexClient` instantiated in [Create a SearchIndexClient](#create-a-searchindexclient)
+`SearchDocument` is the default type returned from queries when you don't
+provide your own.  Here we perform the search, enumerate over the results, and
+extract data using `SearchDocument`'s dictionary indexer.
 
-<!-- embedme ./src/samples/java/com/azure/search/documents/ReadmeSamples.java#L116-L121 -->
-```java
+<!-- embedme ./src/samples/java/com/azure/search/documents/ReadmeSamples.java#L151-L157 -->
+```Java
+SearchPagedIterable searchResultsIterable = searchClient.search("luxury");
+for (SearchResult searchResult: searchResultsIterable) {
+    SearchDocument doc = searchResult.getDocument(SearchDocument.class);
+    String id = (String) doc.get("hotelId");
+    String name = (String) doc.get("hotelName");
+    System.out.println(String.format("This is hotelId %s, and this is hotel name %s.", id, name));
```

Comment:
hotelId --> hotel Id:

Reply:
end of the line %s%n

Reply:
Is `%n` needed here? We use `System.out.println` which definitely start a new line

Reply:
>Is %n needed here? We use System.out.println which definitely start a new line

didn't notice, ignore me!

nit: rather than System.out.println(String.format...) -->  System.out.printf(...) ?